rust/osm2lanes/src/transform/tags_to_lanes/mod.rs

```diff
@@ -363,14 +363,31 @@ fn assemble_ltr(
     })
 }
 
-pub fn unsupported(tags: &Tags, _locale: &Locale, warnings: &mut RoadWarnings) -> ModeResult {
-    if tags.is("highway", "bus_guideway") {
-        return Err(RoadMsg::unimplemented_tag("highway", "bus_guideway").into());
-    }
-
-    if tags.is("highway", "construction") {
-        return Err(RoadMsg::unimplemented_tag("highway", "construction").into());
-    }
+pub fn unsupported(
+    tags: &Tags,
+    _locale: &Locale,
+    warnings: &mut RoadWarnings,
+) -> Result<Highway, RoadError> {
+    let highway = Highway::from_tags(tags);
+    let highway = match highway {
+        Err(None) => return Err(RoadMsg::unsupported_str("way is not highway").into()),
+        Err(Some(s)) => return Err(RoadMsg::unsupported_tag(HIGHWAY, &s).into()),
+        Ok(highway) => match highway {
+            Highway::Road(_)
+                | Highway::Link(_)
+                | Highway::Service
+                | Highway::Residential
+                // Non motorized
+                | Highway::Pedestrian
+                | Highway::Footway
+                | Highway::Cycleway
+                | Highway::Path
+                | Highway::Steps
+                | Highway::Track
+             => highway,
+            highway => return Err(RoadMsg::unimplemented_tag(HIGHWAY, &highway.to_string()).into()),
```

Comment:
What's happening here?

Reply:
if it is not a known supported enum variant, we say it is not implemented.
This replaces the previous construction and guided busway placeholders